util/llbutil/secretprovider/aws_creds.go

```diff
@@ -0,0 +1,116 @@
+package secretprovider
+
+import (
+	"context"
+	"net/url"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/moby/buildkit/session/secrets"
+	"github.com/pkg/errors"
+	"gopkg.in/ini.v1"
+)
+
+// AWSCredentialProvider can load AWS settings from the environment.
+type AWSCredentialProvider struct {
+	mu          sync.Mutex
+	config      *ini.File
+	credsConfig *ini.File
+}
+
+// NewAWSCredentialProvider creates and returns a credential provider for AWS.
+func NewAWSCredentialProvider() *AWSCredentialProvider {
+	return &AWSCredentialProvider{}
+}
+
+// GetSecret attempts to find an AWS secret from either the environment or a local config file.
+func (c *AWSCredentialProvider) GetSecret(ctx context.Context, name string) ([]byte, error) {
+
+	names := map[string]int{
+		"AWS_ACCESS_KEY_ID":     0,
+		"AWS_SECRET_ACCESS_KEY": 0,
+		"AWS_SESSION_TOKEN":     0,
+		"AWS_DEFAULT_REGION":    0,
+	}
+
+	q, err := url.ParseQuery(name)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse secret info")
+	}
+
+	secretName := q.Get("name")
+
+	if _, ok := names[secretName]; ok {
```

Comment:
only using the name might be too fragile, as some users might have a `RUN --secret AWS_ACCESS_KEY_ID`, which would trigger this code if I'm not mistaken?